yt_dlp/extractor/vueschool.py

```diff
@@ -0,0 +1,132 @@
+            vimeo_player_wrapper_dict = extract_attributes(vimeo_wrapper_element)
+            vimeo_id = vimeo_player_wrapper_dict.get(':id')
+        except compat_HTMLParseError:
+            return self.raise_login_required(msg='This video is only available for paid users.', method='cookies')
+
+        description = get_element_by_class('text xl:text-lg', webpage).strip()
+
+        course_dict = self._get_course_info_from_lesson_slug(lesson_slug, id=lesson_slug)
+        course_title = course_dict.get('title')
+        course_slug = course_dict.get('slug')
+
+        chapter_number = None
+        chapter = None
+        lesson_number = None
+        lesson_title = None
+        duration = None
+
+        for chapter_dict in course_dict.get('chapters'):
+            for lesson_dict in chapter_dict.get('lessons'):
+                if lesson_slug == lesson_dict.get('slug'):
+                    chapter_number = int_or_none(chapter_dict.get('course_order') + 1)
+                    chapter = chapter_dict.get('title')
+                    lesson_number = int_or_none(lesson_dict.get('course_order'))
+                    lesson_title = lesson_dict.get('title')
+                    duration = duration
+                    break
+
+        vimeo_url = VimeoIE._smuggle_referrer(f'https://player.vimeo.com/video/{vimeo_id}', url)
+
+        return self.url_result(
+            vimeo_url, VimeoIE, url_transparent=True,
+            video_id=lesson_slug, video_title=lesson_title,
+            id=lesson_slug,
+            series_id=course_slug, series=course_title,
+            chapter_number=chapter_number, chapter=chapter,
+            episode_number=lesson_number, episode=lesson_title,
+            title=lesson_title,
+            description=description,
+            duration=duration
+        )
+
+
+class VueSchoolCourseIE(VueSchoolBaseIE):
+    IE_DESC = 'VueSchool Course'
+    _VALID_URL = r'https?://vueschool\.io/courses/(?P<id>[a-z0-9\-]+)'
+    _TESTS = [{
+        'url': 'https://vueschool.io/courses/vue-js-fundamentals-with-the-composition-api',
+        'info_dict': {
+            'id': 'vue-js-fundamentals-with-the-composition-api',
+            '_type': 'playlist',
+            'title': 'Vue.js 3 Fundamentals with the Composition API',
+        },
+        'params': {
+            'skip_download': True,
+        }
+    }]
+
+    def _real_extract(self, url):
+        course_slug = self._match_id(url)
+        webpage = self._download_webpage(url, video_id=course_slug)
+
+        lesson_slugs = self._html_search_regex(r'https?://vueschool\.io/lessons/(?P<lesson_slug>[0-9a-z\-]+)', webpage, name='lesson_slug')
+
+        course_dict = self._get_course_info_from_lesson_slug(lesson_slugs, id=course_slug)
+        course_title = course_dict.get('title')
+        course_slug = course_dict.get('slug')
+        lessons = self._flatten_lessons_from_chapters(course_dict.get('chapters'))
```

Comment:
```suggestion
        lessons = [x for chapters in course_dict.get('chapters') for x in chapters.get('lessons')]
```

Reply:
List comprehension is used in the latest commit (2da2072).